src/decoder/mod.rs

```diff
@@ -98,6 +97,16 @@ fn read_i64<R: Read + ?Sized>(reader: &mut R) -> DecoderResult<i64> {
     reader.read_i64::<LittleEndian>().map_err(From::from)
 }
 
+/// Placeholder decoder for `Decimal128`. Reads 128 bits and just stores them, does no validation or
+/// parsing.
+#[cfg(not(feature = "decimal128"))]
+#[inline]
+fn read_f128<R: Read + ?Sized>(reader: &mut R) -> DecoderResult<Decimal128> {
+    let mut buf = vec![0u8; 128 / 8];
```

Comment:
Since it's only 16 bytes, it's probably fine to just use an array (which will use the stack rather than the heap):

```rust
let mut buf: [u8; 128 / 8] = [0; 128 / 8];
```

Reply:
done

Reply:
Ah, I didn't even think of doing it in the struct itself too for some reason